### PR TITLE
Initial storage reference docs

### PR DESCRIPTION
**Pull request by tjholm**

### CODE REVIEW

File: reference/storage/bucket.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+---
+**writing**
+
+This permission allows your function to write files to the bucket
+
+---
+**deleting**
+
+This permission allows your function to write files to the bucket
```
Comment (jyecusch) [inline reference/storage/bucket.md:39]: typo - should say it allows the function to *delete* files

File: reference/storage/file-signUrl.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
+Create a signed url for access to file
```
Comment (jyecusch) [inline reference/storage/file-signUrl.md:1]: typo - should this say "access to *a* file"

File: reference/storage/file-signUrl.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+```javascript
+import { bucket, FileMode } from '@nitric/sdk';
+
+const assets = bucket('assets').for('reading', 'writing');
+
+const logo = assets.file('images/logo.png');
+
+const logoUrl = await logo.signUrl(FileMode.Read);
```
Comment (jyecusch) [inline reference/storage/file-signUrl.md:10]: A comment here to highlight the 'read' mode could be useful.

File: reference/storage/file-signUrl.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+Available options are:
+ - `FileMode.Read`
+ - `FileMode.Write`
+
+---
+
+**options** `SignUrlOptions`
+
+Properties
```
Comment (jyecusch) [inline reference/storage/file-signUrl.md:28]: I listed the object properties differently - https://github.com/nitrictech/docs/blob/0e10b06f72daf22a81f79e56300661e68e59fad0/docs/reference/api/api-route.md we should pick one style and use it in both, what do you think?

File: reference/storage/file-signUrl.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+| name   | type     | default | description               |
+| ------ | -------- | ------- | ------------------------- |
+| expiry | `number` | 600     | seconds until link expiry |
+
+---
+
+## Examples
+
+Create a readable link that is valid for the next 10 minutes
+```javascript
```
Comment (jyecusch) [inline reference/storage/file-signUrl.md:39]: Minor thing, there should probably be an extra line between the text and code section - some markdown renderers get funny about this.

File: reference/storage/file-signUrl.md [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+
+const logo = assets.file('images/logo.png');
+
+const logoUrl = await logo.signUrl(FileMode.Read, {
+    expiry: 600
+});
+```
+
+Create a temporary file upload link for a user
+```javascript
```
Comment (jyecusch) [inline reference/storage/file-signUrl.md:52]: add line

File: reference/storage/file-write.md [hunk 7]
```diff
@@ -0,0 +1,20 @@
+Write a file from a bucket
```
Comment (jyecusch) [inline reference/storage/file-write.md:1]: typo - write a file *to* a bucket

**Review by jyecusch — CHANGES_REQUESTED** [review]
Just a couple typos, looks good

File: reference/storage/bucket.md (hunk 1 above)
Comment (tjholm) [reply reference/storage/bucket.md]: done

File: reference/storage/file-signUrl.md (hunk 3 above)
Comment (tjholm) [reply reference/storage/file-signUrl.md:11]: done

File: reference/storage/file-signUrl.md (hunk 5 above)
Comment (tjholm) [reply reference/storage/file-signUrl.md]: done

File: reference/storage/file-signUrl.md (hunk 4 above)
Comment (tjholm) [reply reference/storage/file-signUrl.md]: Updated to your format.

File: reference/storage/file-write.md (hunk 7 above)
Comment (tjholm) [reply reference/storage/file-write.md]: done

File: reference/storage/file-signUrl.md (hunk 6 above)
Comment (tjholm) [reply reference/storage/file-signUrl.md]: done

File: reference/storage/file-signUrl.md (hunk 2 above)
Comment (tjholm) [reply reference/storage/file-signUrl.md]: done

File: reference/storage/file-signUrl.md [hunk 8]
```diff
@@ -0,0 +1,63 @@
+Create a signed url for access to a file
```
Comment (davemooreuws) [inline reference/storage/file-signUrl.md:1]: Change this file name to file-sign-url.md or file-signurl.md

**Review by davemooreuws — CHANGES_REQUESTED** [review]
Looks good, I have a few thoughts.

Should we use snippets for any of these examples or do it later?

Just a change on the signed url file name.

File: reference/storage/file-signUrl.md (hunk 8 above)
Comment (tjholm) [reply reference/storage/file-signUrl.md]: done. I think we can skip snippets for now as we haven't redone them for this new resource centric approach. Can revisit with broader language support I think.

**Review by jyecusch — APPROVED** [review]

**Review by davemooreuws — APPROVED** [review]

**Pull request merged**